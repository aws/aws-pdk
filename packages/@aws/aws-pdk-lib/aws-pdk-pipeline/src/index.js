"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./pdk-pipeline"), exports);
__exportStar(require("./pdk-pipeline-ts-project"), exports);
__exportStar(require("./pdk-pipeline-py-project"), exports);
__exportStar(require("./pdk-pipeline-java-project"), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJpbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7QUFBQSxpREFBK0I7QUFDL0IsNERBQTBDO0FBQzFDLDREQUEwQztBQUMxQyw4REFBNEMiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmVcIjtcbmV4cG9ydCAqIGZyb20gXCIuL3Bkay1waXBlbGluZS10cy1wcm9qZWN0XCI7XG5leHBvcnQgKiBmcm9tIFwiLi9wZGstcGlwZWxpbmUtcHktcHJvamVjdFwiO1xuZXhwb3J0ICogZnJvbSBcIi4vcGRrLXBpcGVsaW5lLWphdmEtcHJvamVjdFwiO1xuIl19