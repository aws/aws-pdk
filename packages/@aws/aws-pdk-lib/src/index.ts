export * as construct from "./construct";
export * as project from "./project";