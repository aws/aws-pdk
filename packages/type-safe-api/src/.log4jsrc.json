{
    "appenders": {
        "out": {
            "type": "stdout",
            "layout": {
                "type": "pattern",
                "pattern": "%[[%d{yyyy-MM-dd-hh:mm:ss}] [%p] [%f{1}] -%] %m"
            }
        }
    },
    "categories": {
        "default": { "appenders": [ "out" ], "level": "INFO", "enableCallStack": true }
    }
}