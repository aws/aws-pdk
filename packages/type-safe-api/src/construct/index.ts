/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./type-safe-rest-api";
export * from "./waf/types";
export * from "./authorizers";
export * from "./integrations";
export * from "./spec";
export * from "./functions";
