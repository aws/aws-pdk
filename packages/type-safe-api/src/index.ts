/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
// instantiate log4js when app is initialized
import { configure as configureLog4js } from "log4js";
import * as log4jsSettings from "./.log4jsrc.json";

configureLog4js(log4jsSettings);

export * from "./construct";
export * from "./project";
