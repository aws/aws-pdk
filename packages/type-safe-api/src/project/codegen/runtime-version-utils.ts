/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { MavenCompileOptions } from "projen/lib/java";
import { JavaVersion, NodeVersion, PythonVersion } from "../languages";

/**
 * Utilities for java runtime versions
 */
class JavaRuntimeVersionUtils {
  /**
   * Get the maven compile options for the given java runtime
   */
  public static getMavenCompileOptions = (
    runtimeVersion?: JavaVersion
  ): MavenCompileOptions => {
    switch (runtimeVersion) {
      case JavaVersion.JAVA_21:
        return { source: "21", target: "21" };
      case JavaVersion.JAVA_17:
        return { source: "17", target: "17" };
      case JavaVersion.JAVA_11:
        return { source: "11", target: "11" };
      case JavaVersion.JAVA_8:
      case undefined: // For backwards compatibility the default source and compile target version is Java 8, running on the Java 17 runtime
        return { source: "1.8", target: "1.8" };
      default:
        throw new Error(`Unsupported runtime version ${runtimeVersion}`);
    }
  };

  /**
   * Return the CDK lambda runtime constant for the given java version
   * @see https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Runtime.html
   */
  public static getLambdaRuntime = (runtimeVersion: JavaVersion): string => {
    switch (runtimeVersion) {
      case JavaVersion.JAVA_8:
        return "JAVA_8_CORRETTO";
      case JavaVersion.JAVA_11:
        return "JAVA_11";
      case JavaVersion.JAVA_17:
        return "JAVA_17";
      case JavaVersion.JAVA_21:
        return "JAVA_21";
      default:
        throw new Error(`Unsupported java runtime ${runtimeVersion}`);
    }
  };
}

/**
 * Utilities for node runtime versions
 */
class NodeRuntimeVersionUtils {
  /**
   * Return the CDK lambda runtime constant for the given node version
   * @see https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Runtime.html
   */
  public static getLambdaRuntime = (runtimeVersion: NodeVersion): string => {
    switch (runtimeVersion) {
      case NodeVersion.NODE_18:
        return "NODEJS_18_X";
      case NodeVersion.NODE_20:
        return "NODEJS_20_X";
      default:
        throw new Error(`Unsupported node runtime ${runtimeVersion}`);
    }
  };

  /**
   * Return the target node version for esbuild
   * @see https://esbuild.github.io/api/#target
   */
  public static getEsbuildNodeTarget = (
    runtimeVersion: NodeVersion
  ): string => {
    switch (runtimeVersion) {
      case NodeVersion.NODE_20:
        return "node20";
      case NodeVersion.NODE_18:
        return "node18";
      default:
        throw new Error(`Unsupported node runtime ${runtimeVersion}`);
    }
  };
}

/**
 * Utilities for python runtime versions
 */
class PythonRuntimeVersionUtils {
  /**
   * Return the CDK lambda runtime constant for the given python version
   * @see https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Runtime.html
   */
  public static getLambdaRuntime = (runtimeVersion: PythonVersion): string => {
    switch (runtimeVersion) {
      case PythonVersion.PYTHON_3_11:
        return "PYTHON_3_11";
      case PythonVersion.PYTHON_3_12:
        return "PYTHON_3_12";
      default:
        throw new Error(`Unsupported python runtime ${runtimeVersion}`);
    }
  };

  /**
   * Return the version string used for a dependency on python
   */
  public static getPythonDependencyVersion = (
    runtimeVersion: PythonVersion
  ): string => {
    return `python@^${PythonRuntimeVersionUtils.getPythonVersionString(
      runtimeVersion
    )}`;
  };

  /**
   * Return the version string used for packaging python lambdas with pip
   */
  public static getPipPackagingPythonVersion = (
    runtimeVersion: PythonVersion
  ): string => {
    return PythonRuntimeVersionUtils.getPythonVersionString(runtimeVersion);
  };

  /**
   * Return the version string for python
   */
  private static getPythonVersionString = (
    runtimeVersion: PythonVersion
  ): string => {
    switch (runtimeVersion) {
      case PythonVersion.PYTHON_3_12:
        return "3.12";
      case PythonVersion.PYTHON_3_11:
        return "3.11";
      default:
        throw new Error(`Unsupported python runtime ${runtimeVersion}`);
    }
  };
}

/**
 * A collection of utilities for runtime versions.
 */
export class RuntimeVersionUtils {
  /**
   * Java utilities
   */
  public static JAVA = JavaRuntimeVersionUtils;
  /**
   * Node utilities
   */
  public static NODE = NodeRuntimeVersionUtils;
  /**
   * Python utilities
   */
  public static PYTHON = PythonRuntimeVersionUtils;
}
