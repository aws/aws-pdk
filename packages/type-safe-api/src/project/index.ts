/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./model/type-safe-api-model-build";
export * from "./model/type-safe-api-async-model-build";
export * from "./model/smithy/smithy-project-definition";
export * from "./model/openapi/open-api-project-definition";
export * from "./type-safe-api-project";
export * from "./model/type-safe-api-model-project";
export * from "./model/smithy/smithy-definition";
export * from "./model/openapi/open-api-definition";
export * from "./type-safe-websocket-api-project";
export * from "./model/type-safe-websocket-api-model-project";
export * from "./model/smithy/smithy-async-definition";
export * from "./model/openapi/open-api-async-definition";
export * from "./model/smithy/types";
export * from "./model/openapi/open-api-model-project";
export * from "./model/openapi/open-api-async-model-project";
export * from "./model/smithy/smithy-async-model-project";
export * from "./model/smithy/smithy-model-project";
export * from "./languages";
export * from "./types";
export * from "./typescript-project-options";
export * from "./python-project-options";
export * from "./java-project-options";
