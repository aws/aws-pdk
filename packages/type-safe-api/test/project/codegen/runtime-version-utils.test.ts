/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import { JavaVersion, NodeVersion, PythonVersion } from "../../../src";
import { RuntimeVersionUtils } from "../../../src/project/codegen/runtime-version-utils";

describe("RuntimeVersionUtils", () => {
  describe("Node", () => {
    it.each(Object.values(NodeVersion))(
      "maps the esbuild target for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.NODE.getEsbuildNodeTarget(version)
        ).toMatchSnapshot();
      }
    );

    it.each(Object.values(NodeVersion))(
      "maps the lambda runtime version for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.NODE.getLambdaRuntime(version)
        ).toMatchSnapshot();
      }
    );
  });

  describe("Java", () => {
    it.each(Object.values(JavaVersion))(
      "maps maven compile options for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.JAVA.getMavenCompileOptions(version)
        ).toMatchSnapshot();
      }
    );

    it.each(Object.values(JavaVersion))(
      "maps the lambda runtime version for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.JAVA.getLambdaRuntime(version)
        ).toMatchSnapshot();
      }
    );
  });

  describe("Python", () => {
    it.each(Object.values(PythonVersion))(
      "maps the pip packaging python version for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.PYTHON.getPipPackagingPythonVersion(version)
        ).toMatchSnapshot();
      }
    );

    it.each(Object.values(PythonVersion))(
      "maps the python dependency version for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.PYTHON.getPythonDependencyVersion(version)
        ).toMatchSnapshot();
      }
    );

    it.each(Object.values(PythonVersion))(
      "maps the lambda runtime version for %s",
      (version) => {
        expect(
          RuntimeVersionUtils.PYTHON.getLambdaRuntime(version)
        ).toMatchSnapshot();
      }
    );
  });
});
