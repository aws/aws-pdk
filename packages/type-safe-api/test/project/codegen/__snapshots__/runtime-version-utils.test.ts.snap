// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuntimeVersionUtils Java maps maven compile options for JAVA_8 1`] = `
{
  "source": "1.8",
  "target": "1.8",
}
`;

exports[`RuntimeVersionUtils Java maps maven compile options for JAVA_11 1`] = `
{
  "source": "11",
  "target": "11",
}
`;

exports[`RuntimeVersionUtils Java maps maven compile options for JAVA_17 1`] = `
{
  "source": "17",
  "target": "17",
}
`;

exports[`RuntimeVersionUtils Java maps maven compile options for JAVA_21 1`] = `
{
  "source": "21",
  "target": "21",
}
`;

exports[`RuntimeVersionUtils Java maps the lambda runtime version for JAVA_8 1`] = `"JAVA_8_CORRETTO"`;

exports[`RuntimeVersionUtils Java maps the lambda runtime version for JAVA_11 1`] = `"JAVA_11"`;

exports[`RuntimeVersionUtils Java maps the lambda runtime version for JAVA_17 1`] = `"JAVA_17"`;

exports[`RuntimeVersionUtils Java maps the lambda runtime version for JAVA_21 1`] = `"JAVA_21"`;

exports[`RuntimeVersionUtils Node maps the esbuild target for NODE_18 1`] = `"node18"`;

exports[`RuntimeVersionUtils Node maps the esbuild target for NODE_20 1`] = `"node20"`;

exports[`RuntimeVersionUtils Node maps the lambda runtime version for NODE_18 1`] = `"NODEJS_18_X"`;

exports[`RuntimeVersionUtils Node maps the lambda runtime version for NODE_20 1`] = `"NODEJS_20_X"`;

exports[`RuntimeVersionUtils Python maps the lambda runtime version for PYTHON_3_11 1`] = `"PYTHON_3_11"`;

exports[`RuntimeVersionUtils Python maps the lambda runtime version for PYTHON_3_12 1`] = `"PYTHON_3_12"`;

exports[`RuntimeVersionUtils Python maps the pip packaging python version for PYTHON_3_11 1`] = `"3.11"`;

exports[`RuntimeVersionUtils Python maps the pip packaging python version for PYTHON_3_12 1`] = `"3.12"`;

exports[`RuntimeVersionUtils Python maps the python dependency version for PYTHON_3_11 1`] = `"python@^3.11"`;

exports[`RuntimeVersionUtils Python maps the python dependency version for PYTHON_3_12 1`] = `"python@^3.12"`;
